pallets/msa/src/offchain_storage.rs

```diff
@@ -69,6 +75,30 @@ pub const RPC_FINALIZED_BLOCK_REQUEST_URL: &str = "http://localhost:9944";
 pub const RPC_FINALIZED_BLOCK_REQUEST_BODY: &[u8; 78] =
 	b"{\"id\": 10, \"jsonrpc\": \"2.0\", \"method\": \"chain_getFinalizedHead\", \"params\": []}";
 
+/// The overarching Offchain replay type that can allow replay of different events across different pallets
```

Comment:
Why is this in here?

To have a better ergonomics when calling the extrinsic from PolkadotJs UI it makes sense to use `T: Config` generic and that is not available inside `common/primitives`. We can always replace the generics and move them out in future if the tradeoff changes.